### PR TITLE
[SUCE-147] Generate HTMLs from PO data

### PR DESCRIPTION
**Pull request by gradam**

### CODE REVIEW

File: server/server/data_loader/po.py [hunk 1]
```diff
@@ -1,60 +1,66 @@
...
     markup = []
     msgids = []
     msgstrs = []
-    
+
     po = polib.pofile(po_file)
-    
+
     for entry in po:
         ctx = entry.msgctxt
```
Comment (TrebuhD) [inline server/server/data_loader/po.py]: is `ctx` used anywhere?

File: server/server/data_loader/po.py [hunk 2]
```diff
@@ -194,3 +200,33 @@ def load_po_texts(change_tracker, po_dir, db):
...
                     collection = db.collection('po_strings')
                     insert_or_update(collection, doc)
+
+    generate_htmls(db)
+
+
+def generate_html(markup: str, translation: List[List[str]]) -> str:
+    translator = {string[0]: string[1] for string in translation}
+
+    pattern = re.compile('|'.join(re.escape(f'{key}">') for key in translator.keys()))
```
Comment (TrebuhD) [inline server/server/data_loader/po.py]: it's not immediately obvious what's happening here, do you think we should add comments / docstrings?

File: server/server/migrations/migrations/add_po_htmls_collection_005.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+class SecondMigration(Migration):
+    migration_id = 'add_po_htmls_collection_005'
+    tasks = ['create_collection']
+
+    def create_collection(self):
+        db = get_db()
+
+        po_htmls = db.create_collection('po_htmls')
+        po_htmls.add_hash_index(fields=['uid'], unique=False)
```
Comment (TrebuhD) [inline server/server/migrations/migrations/add_po_htmls_collection_005.py]: shouldn't the uid be unique?

Comment (gradam) [reply server/server/migrations/migrations/add_po_htmls_collection_005.py]: It is uid of sutta. we can have multiple translations of a single sutta.

Comment (TrebuhD) [reply server/server/migrations/migrations/add_po_htmls_collection_005.py]: ah, I see.

File: server/server/data_loader/po.py [hunk 4]
```diff
@@ -205,6 +203,22 @@ def load_po_texts(change_tracker, po_dir, db):
 
 
 def generate_html(markup: str, translation: List[List[str]]) -> str:
+    """ Takes HTML markup and insert translated string between sc-seg tags with id matching text uid.
```
Comment (TrebuhD) [inline server/server/data_loader/po.py]: https://www.python.org/dev/peps/pep-0257/

`The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".`

Comment (gradam) [reply server/server/data_loader/po.py]: You have all that information here. It is Google's style doc-string

Comment (TrebuhD) [reply server/server/data_loader/po.py]: You use both a descriptive style (`Takes`) and command style (`insert`) here. Is that Google style? :)

**Pull request merged**